include/spi.h

```diff
@@ -152,25 +201,33 @@ struct spi_cs_control {
  *     mode                [ 1 : 3 ]   - Polarity, phase and loop mode.
  *     transfer            [ 4 ]       - LSB or MSB first.
  *     word_size           [ 5 : 10 ]  - Size of a data frame in bits.
- *     lines               [ 11 : 12 ] - MISO lines: Single/Dual/Quad.
+ *     lines               [ 11 : 12 ] - MISO lines: Single/Dual/Quad/Octal.
  *     cs_hold             [ 13 ]      - Hold on the CS line if possible.
  *     lock_on             [ 14 ]      - Keep resource locked for the caller.
- *     eeprom              [ 15 ]      - EEPROM mode.
+ *     cs_active_high      [ 15 ]      - Active high CS logic.
  * @param slave is the slave number from 0 to host controller slave limit.
  * @param cs is a valid pointer on a struct spi_cs_control is CS line is
  *    emulated through a gpio line, or NULL otherwise.
  *
- * @note cs_hold, lock_on and eeprom_rx can be changed between consecutive
- * transceive call.
+ * @note Only cs_hold and lock_on can be changed between consecutive
+ * transceive call. Rest of the attributes are not meant to be tweaked.
+ * (I.e.: one struct spi_config is necessary per-device on the bus)
```

Comment:
Once more regarding not changing the other fields in `operation`.
Of course, a case when one needs to use different communication settings on the same bus can be handled with several `spi_config` structures. But this can be an unnecessary waste of memory, and it may be significant in some cases. The whole bus configuration is in `u16_t` field, so it can be easily stored in the driver and compared on the next `transceive` call to see if something actually needs to be reconfigured in the controller. The need of reconfiguration must be checked also in the current version, because each `transceive` call may be done with a different `spi_config` structure than the previous one, and it may (but also may not) contain a completely different configuration.
The similar story is with forcing usage of a separate `spi_config` structure for each device on the bus. When there are many slaves on the bus and all use the same communication settings, one could use one `spi_config` structure and change only the `cs` (or `slave`) field in it accordingly. And I can't see a strong reason why to forbid this.

Reply:
Currently configuration change verification is only done on pointer comparison.
I understand your point, but apart from very highly specific use case: nobody is going to change one spi_config operation to access different devices on the bus. 
You are not going to be able to use one spi_config for different drivers, you are pointing out the slave part, but it works the same for master: when you do a driver for X on subsystem K you have your own spi_config. At built time your board enables Y and Z on subsystems M and N respectively, which also access the same spi bus and therefore have a dedicated spi_config. And let's say they have exactly same configurations bits. How are you going to merge these 3 _properly_ so it's transparent for both the device driver dev and the user who will play with Kconfig/dts, without runtime necessary functions etc...? We had this idea in the very beginning of Zephyr and it just ended up impracticable path: too much hassle for such a little tiny gain, if any actually. And I am not even putting the case were spi_cs_control would be different for each devices in the picture.

Plus, I am planning to switch all spi_config to const, everywhere so it'll go on ROM and that's it. And I expect all other APIs to follow the same, where relevant. This will need a change in the device driver model, which I am prototyping, so let's see this in near future. That's out of this PR scope.

Reply:
I still consider it (the necessity to provide a separate `spi_config` structure for each device on the bus and for each different configuration of the bus) as an unnecessary limitation. But let's keep it as it is now for this PR, and maybe discuss it later in the context of improvements in the device driver model.